workflows/templates/qdm-preprocess.yaml

```diff
@@ -139,11 +139,11 @@ spec:
         parameters:
           - name: out-zarr
             valueFrom:
-              expression: "inputs.parameters['correct-wetday-frequency'] == 'true' ? tasks['correct-wetday-frequency'].outputs.parameters['out-zarr'] : inputs.parameters['in-zarr']"
+              expression: "inputs.parameters['correct-wetday-frequency'] == 'true' ? tasks['wetday-frequency-precorrection'].outputs.parameters['out-zarr'] : inputs.parameters['in-zarr']"
```

Comment:
So this lets this step return the plain input it received it if has nothing to do (`correct-wetday-frequency== false`). 

This trick was already there but I read it in detail just now. Nice !

Reply:
@emileten Yeah, it's called "conditional parameters". A relatively new feature https://argoproj.github.io/argo-workflows/conditional-artifacts-parameters/#conditional-parameters

I'm trying not to abuse it because it is a bit intimidating and hard to read, IMHO.